### DISCUSSION TITLE
LLM providers supported by PyMuPDF4LLM

### DISCUSSION THREAD

**Discussion by yasuyukitera** [pymupdf/pymupdf4llm#180] [Q&A] [answered]
Does PyMuPDF4LLM plan to support LLM providers other than OpenAI?

**Comment by JorjMcKie** [comment, marked as answer]
We will not scan the market for such providers actively.
Our current direct support for LlamaIndex was a friendly contribution (PR) by one of our users.
We are open to consider / accept other, similar contributions.

**Reply by yasuyukitera** [reply]
Thank you for answering my question.

By the way, I am considering modifying PyMuPDF4LLM so that it can be used against Amazon Bedrock. Right now, PyMuPDF4LLM is OpenAI-only, so I'm looking for a place that calls the OpenAI API, but I can't find it. I am sorry, but could you please tell me how it works?

**Reply by JorjMcKie** [reply]
Best look at the source that accesses LlamaIndex in [this](https://github.com/pymupdf/RAG/blob/main/pymupdf4llm/pymupdf4llm/llama/pdf_markdown_reader.py) script.

**Reply by yasuyukitera** [reply]
Thank you very much. It helped me a lot. I read the source code from the points you taught me.

I seem to have made a big mistake.
I see that PyMuPDF4LLM calls only PyMuPDF when converting a document to markdown format, and does not use an external LLM (e.g. GPT-4).So PyMuPDF4LLM is just a tool to convert to LLM-friendly markdown format.

**Reply by JorjMcKie** [reply]
Exactly. PyMuPDF / PyMuPDF4LLM contain no external dependencies in this respect.
